lib/Dialect/Handshake/Transforms/Buffers.cpp

```diff
@@ -111,17 +111,31 @@ struct HandshakeInsertBuffersPass
     }
   };
 
-  // Perform a depth first search and add a buffer to any un-buffered channel
-  // where it makes reasonable sense.
-  void bufferAllStrategy(handshake::FuncOp f, OpBuilder &builder,
-                         unsigned numSlots, bool sequential = true) {
+  // Add a buffer to any un-buffered channel.
+  void bufferAllStrategy(
+      handshake::FuncOp f, OpBuilder &builder, unsigned numSlots,
+      bool sequential = true,
+      llvm::function_ref<bool(Operation *, Operation *)> filter =
+          [](auto *, auto *) { return false; }) {
 
     for (auto &arg : f.getArguments()) {
       if (!shouldBufferArgument(arg))
         continue;
-      for (auto &use : arg.getUses())
-        insertBufferRecursive(use, builder, numSlots, isUnbufferedChannel,
-                              /*sequential=*/sequential);
+      insertBuffer(arg.getLoc(), arg, builder, numSlots, sequential);
+    }
+
+    for (auto &defOp : f.getOps()) {
+      for (auto res : defOp.getResults()) {
+        for (auto useOp : res.getUsers()) {
+          if (!isUnbufferedChannel(&defOp, useOp))
+            continue;
+
+          if (filter(&defOp, useOp))
```

Comment:
I believe this is the only place `filter` is used, and it is always returning false. Can it be removed?